stan/math/rev/functor/apply_scalar_binary.hpp

```diff
@@ -0,0 +1,160 @@
+ */
+template <typename T1, typename T2, typename F,
+          require_var_matrix_t<T1>* = nullptr,
+          require_std_vector_vt<std::is_integral, T2>* = nullptr>
+inline auto apply_scalar_binary(const T1& x, const T2& y, const F& f) {
+  check_matching_sizes("Binary function", "x", x, "y", y);
+  return f(x, y);
+}
+
+/**
+ * Specialisation for use with a one-dimensional std::vector of integer types
+ * and one `var_value<Eigen vector>` (row or column).
+ *
+ * @tparam T1 Type of first argument to which functor is applied.
+ * @tparam T2 Type of second argument to which functor is applied.
+ * @tparam F Type of functor to apply.
+ * @param x Integer std::vector input to which operation is applied.
+ * @param y Eigen input to which operation is applied.
+ * @param f functor to apply to inputs.
+ * @return Eigen object with result of applying functor to inputs.
+ */
+template <typename T1, typename T2, typename F,
+          require_std_vector_vt<std::is_integral, T1>* = nullptr,
+          require_var_matrix_t<T2>* = nullptr>
+inline auto apply_scalar_binary(const T1& x, const T2& y, const F& f) {
+  check_matching_sizes("Binary function", "x", x, "y", y);
+  return f(x, y);
+}
+
+/**
+ * Specialisation for use with one `var_value<Matrix>` and
+ * a two-dimensional std::vector of integer types
+ *
+ * @tparam T1 Type of first argument to which functor is applied.
+ * @tparam T2 Type of second argument to which functor is applied.
+ * @tparam F Type of functor to apply.
+ * @param x var with Eigen matrix inner type to which operation is applied.
+ * @param y Nested integer std::vector input to which operation is applied.
+ * @param f functor to apply to inputs.
+ * @return Eigen object with result of applying functor to inputs.
+ */
+template <typename T1, typename T2, typename F,
+          require_var_matrix_t<T1>* = nullptr,
+          require_std_vector_vt<is_std_vector, T2>* = nullptr,
+          require_std_vector_st<std::is_integral, T2>* = nullptr>
+inline auto apply_scalar_binary(const T1& x, const T2& y, const F& f) {
+  return f(x, y);
+}
+
+/**
+ * Specialisation for use with a two-dimensional std::vector of integer types
+ * and one `var_value<Matrix>`.
+ *
+ * @tparam T1 Type of first argument to which functor is applied.
+ * @tparam T2 Type of second argument to which functor is applied.
+ * @tparam F Type of functor to apply.
+ * @param x Nested integer std::vector input to which operation is applied.
+ * @param y var value with inner Eigen matrix input to which operation is
+ * applied.
+ * @param f functor to apply to inputs.
+ * @return Eigen object with result of applying functor to inputs.
+ */
+template <typename T1, typename T2, typename F,
+          require_std_vector_vt<is_std_vector, T1>* = nullptr,
+          require_std_vector_st<std::is_integral, T1>* = nullptr,
+          require_var_matrix_t<T2>* = nullptr>
+inline auto apply_scalar_binary(const T1& x, const T2& y, const F& f) {
+  return f(x, y);
+}
```

Comment:
```suggestion
template <typename T1, typename T2, typename F,
          require_all_not_matrix_t<T1, T2>* = nullptr,
          require_all_not_std_vector_vt<std::is_integral, T1, T2>* = nullptr,
          require_any_var_matrix_t<T1, T2>* = nullptr>
inline auto apply_scalar_binary(const T1& x, const T2& y, const F& f) {
  return f(x, y);
}
```

I \*think\* this covers all of the overloads which don't have `dim` or `size` checking

Reply:
I couldn't get this to work :-/. But I was able to cut out half of the ones without dim or size checking by using `require_any_*`